component/common/src/main/java/org/exoplatform/platform/common/rest/services/WhoIsOnLinePortlet/WhoIsOnLineRestService.java

```
@@ -0,0 +1,140 @@
+            userId = ConversationState.getCurrent().getIdentity().getUserId();
+        }
+        if (StringUtils.isBlank(userId)) {
+            return Response.ok(jsonArray, MediaType.APPLICATION_JSON).build();
+        }
+        Space space = null;
+        if (StringUtils.isNotBlank(spaceName)) {
+            space = spaceService.getSpaceByPrettyName(spaceName);
+        }
+        if (space != null) {
+            jsonArray = getOnlineUsersMembersOfSpace(userId, space);
+        } else {
+            jsonArray = getOnlineUsers(userId);
+        }
+        return Response.ok(jsonArray, MediaType.APPLICATION_JSON).build();
+    }
+
+    private JSONArray getOnlineUsersMembersOfSpace(String userId, Space space) {
+        JSONArray jsonArray = new JSONArray();
+        try {
+            List<UserStateModel> users = userStateService.online();
+            Collections.reverse(users);
+            if (users.size() > MAX_USER) {
+                users = users.subList(0, INDEX_USER);
+            }
+            String superUserName = System.getProperty("exo.super.user");
+            for (UserStateModel userModel : users) {
+                String user = userModel.getUserId();
+                if (user.equals(userId) || user.equals(superUserName) || INVISIBLE.equals(userModel.getStatus()) || !spaceService.isMember(space,user)) continue;
+                JSONObject json = new JSONObject();
+                Identity userIdentity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, user,false);
+                Profile userProfile = userIdentity.getProfile();
+                json.put("id", userProfile.getId());
+                json.put("profileUrl", LinkProvider.getUserProfileUri(userIdentity.getRemoteId()));
+                json.put("avatar", LinkProviderUtils.getUserAvatarUrl(userProfile));
+                jsonArray.put(json);
+            }
+            return jsonArray;
+
+        } catch (Exception e) {
+            return jsonArray;
```

The exception should not be swallowed and if something went wrong we should instead return an HTTP error.